### PR TITLE
Fix issues seen during delta-rs integration

### PR DESCRIPTION
**Pull request by yjshen**

### CODE REVIEW

File: hdfs/src/object_store/hdfs.rs [hunk 1]
```diff
@@ -120,7 +120,7 @@ impl ObjectStore for HadoopFileSystem {
         let location = HadoopFileSystem::path_to_filesystem(location);
 
         maybe_spawn_blocking(move || {
-            let file = match hdfs.create(&location) {
+            let file = match hdfs.create_with_overwrite(&location, true) {
```
Comment (yjshen) [inline hdfs/src/object_store/hdfs.rs:123]: To align with the semantics of LocalFileSystem's 'put' function, please refer to https://github.com/apache/arrow-rs/blob/master/object_store/src/local.rs#L608.

File: hdfs/src/object_store/hdfs.rs [hunk 2]
```diff
@@ -229,7 +229,8 @@ impl ObjectStore for HadoopFileSystem {
         &self,
         prefix: Option<&Path>,
     ) -> object_store::Result<BoxStream<'_, object_store::Result<ObjectMeta>>> {
-        let prefix = prefix.expect("Prefix for hdfs should not be None");
+        let default_path = Path::from(self.get_path_root());
+        let prefix = prefix.unwrap_or(&default_path);
```
Comment (yjshen) [inline hdfs/src/object_store/hdfs.rs:233]: To align with the semantics of LocalFileSystem's 'list' function, which provides a default root when a prefix is not given. Please refer to https://github.com/apache/arrow-rs/blob/master/object_store/src/local.rs#L437-L440.

File: hdfs/src/object_store/hdfs.rs [hunk 3]
```diff
@@ -284,7 +285,8 @@ impl ObjectStore for HadoopFileSystem {
     /// List files and directories directly under the prefix path.
     /// It will not recursively search leaf files whose depth is larger than 1
     async fn list_with_delimiter(&self, prefix: Option<&Path>) -> object_store::Result<ListResult> {
-        let prefix = prefix.expect("Prefix for hdfs should not be None");
+        let default_path = Path::from(self.get_path_root());
+        let prefix = prefix.unwrap_or(&default_path);
```
Comment (yjshen) [inline hdfs/src/object_store/hdfs.rs:289]: ditto

File: hdfs/src/object_store/hdfs.rs [hunk 4]
```diff
@@ -399,7 +401,7 @@ impl ObjectStore for HadoopFileSystem {
 /// Create Path without prefix
 pub fn get_path(full_path: &str, prefix: &str) -> Path {
     let partial_path = &full_path[prefix.len()..];
-    Path::from(partial_path)
+    Path::parse(partial_path).unwrap()
```
Comment (yjshen) [inline hdfs/src/object_store/hdfs.rs:404]: To avoid double encoding the path.

File: hdfs/Cargo.toml [hunk 5]
```diff
@@ -17,7 +17,7 @@
 
 [package]
 name = "datafusion-objectstore-hdfs"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (yjshen) [inline hdfs/Cargo.toml:20]: The current master hasn't been updated I think.

File: hdfs/src/object_store/hdfs.rs (hunk 1 above)
Comment (yahoNanJing) [reply hdfs/src/object_store/hdfs.rs:123]: Thanks for pointing this out

File: hdfs/src/object_store/hdfs.rs (hunk 2 above)
Comment (yahoNanJing) [reply hdfs/src/object_store/hdfs.rs:233]: Agree. 👍

File: hdfs/Cargo.toml (hunk 5 above)
Comment (yahoNanJing) [reply hdfs/Cargo.toml:20]: my bad. I forgot to update it.

**Comment by yahoNanJing** [comment]
LGTm

**Pull request merged**